### PR TITLE
[Resolves #1264] Update AWS CDK Example for CDK v2 and Asset Support

### LINKED ISSUE #1264: The Sceptre Template AWS CDK Example is out of date and does not support CDK Assets
### Subject of the issue

The Sceptre template AWS CDK example in https://github.com/Sceptre/sceptre/blob/master/docs/_source/docs/templates.rst is out of date, as it is for AWS CDK v1 rather than the current v2 and it does not support CDK stacks that require assets (constructs that create CloudFormation custom resources for example). 

### Your environment
* version of sceptre `3.2.0`
* version of python `3.8.8`
* which OS/distro: `Windows 10`
* AWS CDK version: `2.51.1`

### Steps to reproduce

Use the following code with the current AWS CDK example. The `BucketDeployment` construct creates a custom CloudFormation resource:

```python
        s3_bucket = s3.Bucket(self,
                              "S3Bucket",
                              bucket_name=sceptre_user_data['BucketName']
                              )
 
        cdk.aws_s3_deployment.BucketDeployment(self,
                                               "S3Deployment",
                                               sources=[cdk.aws_s3_deployment.Source.data(
                                                   'object-key.txt', 'hello, world!')],
                                               destination_bucket=s3_bucket
                                               )
```
 
### Expected behaviour

The CDK stack should be deployed successfully,

### Actual behaviour

`sceptre launch` fails with the following error:
```
ImportError: cannot import name 'core' from 'aws_cdk'
```

Once the example code has been refactored for CDK v2.x `sceptre launch` fails with the following error:
```
[2022-11-23 16:52:08] - dev/cdk/s3/s3-original S3DeploymentAwsCliLayer8AAFE44F AWS::Lambda::LayerVersion CREATE_FAILED Error occurred while GetObject. S3 Error Code: NoSuchKey. S3 Error Message: The specified key does not exist. (Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException; Request ID: 370f2978-891f-428d-921e-1f7116bacc19; Proxy: null)
```

This is due to the CDK assets for the CloudFormation custom resource not having been published.

### Solution

I have an AWS CDK v2 example for Sceptre that published the CDK assets and I am happy to raise a PR to update the current example with it.

### PR DESCRIPTION
**Pull request by X-Guardian**
Updates the AWS CDK example for CDK v2 and adds asset support.

- Resolves #1264 

## PR Checklist

- [ ] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [ ] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [ ] All unit tests (`make test`) are passing.
- [ ] Used the same coding conventions as the rest of the project.
- [ ] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [ ] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

File: docs/_source/docs/templates.rst [hunk 1]
```diff
@@ -192,51 +192,230 @@ AWS CDK
...
+    template = app_synth.get_stack_by_name(stack_name).template
+    return yaml.safe_dump(template)
+
+.. _AWS CDK: https://github.com/aws/aws-cdk
 
 
 Generate cloudformation:
 
-``sceptre generate dev/S3CdkStack.py``
+``sceptre generate dev/S3CdkStack.yaml``
```
[the remark this answers is not in the document]
Comment (X-Guardian) [reply docs/_source/docs/templates.rst:283]: No, you need to generate the Sceptre stack config file that references the template.

File: docs/_source/docs/templates.rst [hunk 2]
```diff
@@ -192,51 +192,230 @@ AWS CDK
...
 
 .. code-block:: yaml
 
+  Parameters:
```
[the remark this answers is not in the document]
Comment (X-Guardian) [reply docs/_source/docs/templates.rst:287]: Hehe, I thought it was the point to show this.

**Comment by X-Guardian** [comment]
I've also created an initial version of a Sceptre CDK template handler, as per your suggestion. You can take a look at it here: https://github.com/X-Guardian/sceptre-cdk-handler/pull/1

File: docs/_source/docs/templates.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfalkenstein) [reply docs/_source/docs/templates.rst:283]: Oh, 🤦 . That's right. Whoops!

File: docs/_source/docs/templates.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (jfalkenstein) [reply docs/_source/docs/templates.rst:287]: Nah. This is too much stuff without much gain. I don't think we need to include the sample generated template here. I just don't see what the reader gains from this.

**Comment by jfalkenstein** [comment]
I like where you're going and we definitely need to update these docs, but let's hold off on changing these docs until the template handler is available, and then we can reference the CDK handler in the docs rather than including this rather tricky code sample.

File: docs/_source/docs/templates.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (X-Guardian) [reply docs/_source/docs/templates.rst]: OK, removed the generated template

[the remark this answers is not in the document]
Comment (jfalkenstein) [reply docs/_source/docs/templates.rst]: Thanks!

**Comment by jfalkenstein** [comment]
I'm applying the "On Hold" label to this PR until we can get the Template Handler PR merged/deployed. The instructions will be a lot simpler when we can just say "Use this template handler".

**Comment by jfalkenstein** [comment]
@X-Guardian, the CDK Handler has now been deployed! You can update the docs here to point to it.

**Comment by jfalkenstein** [comment]
Closing this as a duplicate of #1285

**Pull request closed without merging**